### PR TITLE
Add support for understanding arm64 libraries and components

### PR DESCRIPTION
**Pull request by jbevain**
Tentative fix for #3746.

With this, the repro project of:

https://developercommunity.visualstudio.com/content/problem/330497/cannot-build-uwp-app-for-arm64-that-references-arm.html

Builds.

### CODE REVIEW

**Comment by jbevain** [comment]
The lack of entry for `ARM64` in `System.Reflection.ProcessorArchitecture` forces the use of a constant.
We're using the same value that Roslyn and `System.Reflection.Metadata` are using.

File: src/Tasks/AssemblyDependency/ReferenceTable.cs [hunk 1]
```diff
@@ -2798,6 +2798,9 @@ private bool VerifyArchitectureOfImplementationDll(string dllPath, string winmdF
                     case NativeMethods.IMAGE_FILE_MACHINE_ARMV7:
                         dllArchitecture = SystemProcessorArchitecture.Arm;
                         break;
+                    case NativeMethods.IMAGE_FILE_MACHINE_ARM64:
+                        dllArchitecture = (SystemProcessorArchitecture) 6; // There's no entry for ARM64 in SystemProcessorArchitecture, use the next available constant
```
[the remark this answers is not in the document]
Comment (jbevain) [reply src/Tasks/AssemblyDependency/ReferenceTable.cs]: I'm happy to bump to a value that .NET doesn't know how to represent (> 7).

I've started a discussion on `CoreCLR` that's linked below to see if they have any recommendation on the approach here, and if it makes sense to add that enum value going forward.

**Comment by jbevain** [comment]
@rainersigwald regardless of the CoreCLR discussion, is this good to go? Could we get that in 16.2?

**Comment by rainersigwald** [comment]
Thanks @jbevain. My plan is to get an updated MSBuild insertion into VS with this commit ~tomorrow; I'll let you know.

**Comment by jbevain** [comment]
@rainersigwald 🙇

**Pull request merged**